src/openfl/display/Shader.hx

```diff
@@ -417,79 +343,78 @@ class Shader {
 			
 		}
 		
+		var textureIndex = 0;
 		while (regex.matchSub (source, lastMatch)) {
 			
 			type = regex.matched (1);
 			name = regex.matched (2);
 			
 			if (StringTools.startsWith (type, "sampler")) {
 				
-				var input = new ShaderInput<BitmapData> ();
-				input.name = name;
+				var input = new ShaderParameterSampler (name, textureIndex);
+				textureIndex++;
 				__inputBitmapData.push (input);
 				Reflect.setField (data, name, input);
 				
 			} else {
 				
-				var parameterType:ShaderParameterType = switch (type) {
-					
-					case "bool": BOOL;
-					case "double", "float": FLOAT;
-					case "int", "uint": INT;
-					case "bvec2": BOOL2;
-					case "bvec3": BOOL3;
-					case "bvec4": BOOL4;
-					case "ivec2", "uvec2": INT2;
-					case "ivec3", "uvec3": INT3;
-					case "ivec4", "uvec4": INT4;
-					case "vec2", "dvec2": FLOAT2;
-					case "vec3", "dvec3": FLOAT3;
-					case "vec4", "dvec4": FLOAT4;
-					case "mat2", "mat2x2": MATRIX2X2;
-					case "mat2x3": MATRIX2X3;
-					case "mat2x4": MATRIX2X4;
-					case "mat3x2": MATRIX3X2;
-					case "mat3", "mat3x3": MATRIX3X3;
-					case "mat3x4": MATRIX3X4;
-					case "mat4x2": MATRIX4X2;
-					case "mat4x3": MATRIX4X3;
-					case "mat4", "mat4x4": MATRIX4X4;
-					default: null;
-					
-				}
-				
-				switch (parameterType) {
-					
-					case BOOL, BOOL2, BOOL3, BOOL4:
-						
-						var parameter = new ShaderParameter<Bool> ();
-						parameter.name = name;
-						parameter.type = parameterType;
-						__paramBool.push (parameter);
-						Reflect.setField (data, name, parameter);
-					
-					case INT, INT2, INT3, INT4:
+				var parameter:ShaderParameter;
+				if (!isUniform) {
+					parameter = new ShaderParameterAttrib(name);
+				} else {
+					var parameterType:ShaderParameterType = switch (type) {
```

Comment:
Can't we create the ShaderParameter directly from this switch and so remove the second one ?

Reply:
I think so, I was just lazy to change that. Will do!